main.nf

```diff
@@ -230,13 +232,74 @@ if (params.hashes){
       .splitText()
       .map{ row -> tuple(row.replaceAll("\\s+", ""), "hash" )}
       .transpose()
-      .into { ch_hashes_to_group_for_joining; ch_hashes_to_group_for_hash2kmer }
+      .into { ch_hash_to_group_for_joining_after_hash2kmer;
+        ch_hash_to_group_for_joining_after_hash2sig;
+        ch_hash_to_group_for_hash2kmer;
+        ch_hash_to_group_for_hash2sig
+       }
 
-  ch_hashes_to_group_for_hash2kmer
+  ch_hash_to_group_for_hash2kmer
     .map{ it -> it[0] }
     .set{ ch_hashes_for_hash2kmer }
 }
 
+
+// Utility functions for sanitizing output
+def groupCleaner(group) {
+  return group.replaceAll(' ', '_').replaceAll('/', '-slash-').toLowerCase()
+}
+
+def hashCleaner(hash) {
+  return hash.replaceAll('\\n', '')
+}
+
+////////////////////////////////////////////////////
+/* --         Parse gene counting       -- */
+////////////////////////////////////////////////////
+if (params.csv_has_is_aligned) {
+  if (params.csv) {
+    // Provided a csv file mapping sample_id to protein fasta path
+    // Channel
```

Comment:
can this commented out code be removed?

csv_has_is_aligned? doesn't sound too intuitive and has no docs in help